lib/api.js

```diff
@@ -301,7 +302,8 @@ function normalizeHTML(html, mimeType) {
       defaultEncoding: mimeType.isXML() ? "UTF-8" : "windows-1252",
       transportLayerEncodingLabel: mimeType.parameters.get("charset")
     });
-    html = whatwgEncoding.decode(html, encoding);
+    html = legacyHookDecode(html, encoding);
+    encoding = toCasedEncodingName(encoding);
```

Comment:
Remove this since per previous discussion we'll store the canonicalized encoding labels.